src/Integration.UnitTests/Suppression/SuppressedIssueMatcherTests.cs

```diff
@@ -0,0 +1,139 @@
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Integration.Suppression;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Suppression
+{
+    [TestClass]
+    public class SuppressedIssueMatcherTests
+    {
+        private SuppressedIssueMatcher testSubject;
+        private Mock<ISonarQubeIssuesProvider> mockIssuesProvider;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            mockIssuesProvider = new Mock<ISonarQubeIssuesProvider>();
+            testSubject = new SuppressedIssueMatcher(mockIssuesProvider.Object);
+        }
+
+        [TestMethod]
+        public void Ctor_NullProvider_Throws()
+        {
+            Action act = () => new SuppressedIssueMatcher(null);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("issuesProvider");
+        }
+
+        [TestMethod]
+        public void MatchExists_NullIssue_Throws()
+        {
+            Action act = () => testSubject.SuppressionExists(null);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("issue");
+        }
+
+        [DataTestMethod]
+        [DataRow("RightRuleId", 1, "RightHash", true)]    // exact matches
+        [DataRow("rightRULEID", 1, "RightHash", true)]    // rule-id is case-insensitive
+        [DataRow("RightRuleId", 1, "wrong hash", true)]   // matches on line
+        [DataRow("RightRuleId", 9999, "RightHash", true)] // matches on hash only
+
+        [DataRow("RightRuleId", 2, "righthash", false)]   // hash is case-sensitive
+        [DataRow("RightRuleId", 2, "wrong hash", false)]  // wrong line and hash
+        [DataRow("wrong rule Id", 1, "RightHash", false)]
+        public void MatchExists_SingleServerIssue(string serverRuleId, int serverIssueLine, string serverHash, bool expectedResult)
+        {
+            var issueToMatch = CreateIssueToMatch("RightRuleId", 1, "RightHash");
+            ConfigureServerIssues(issueToMatch, CreateServerIssue(serverRuleId, serverIssueLine, serverHash));
+
+            // Act and assert
+            testSubject.SuppressionExists(issueToMatch).Should().Be(expectedResult);
+        }
+
+        [TestMethod]
+        public void MatchExists_NoServerIssues_ReturnsFalse()
+        {
+            // Arrange
+            var issueToMatch = CreateIssueToMatch("rule1", 1, "hash1");
+            ConfigureServerIssues(issueToMatch, Array.Empty<SonarQubeIssue>());
+
+            // Act and assert
+            testSubject.SuppressionExists(issueToMatch).Should().BeFalse();
+        }
+
+        [DataTestMethod]
+        [DataRow("aaa", 222, "aaa hash", true)]
+        [DataRow("bbb", 333, "bbb hash", true)]
+        [DataRow("ccc", 444, "ccc hash", true)]
+        [DataRow("xxx", 111, "xxx hash", false)]
+        public void MatchExists_MultipleServerIssues_NoMatches_ReturnsFalse(string localRuleId, int localIssueLine, string localHash, bool expectedResult)
```

Comment:
Is the test name correct? it says "returns false" but most cases actually return true.
Maybe it should be MatchExists_MultipleServerIssues ?

Reply:
Yep, it should be `...MultipleServerIssues`